### ISSUE TITLE
Refactor representation of nodes in the AST

### ISSUE THREAD

**Issue by jan-christiansen** [sagr4019/ResearchProject#108]
In order to represent the AST node ```if e then c else c```, either use

```python
{ 'If' :
   { 'Condition' : e
   , 'Then' : c
   , 'Else' : c
   }
}
```

or

```python
{ 'Kind' : 'If'
, 'Condition' : e
, 'Then' : c
, 'Else' : c
}
```

**Comment by MadeInGer** [comment]
Done in https://github.com/sagr4019/ResearchProject/commit/0a877910207f74aaf39ead367f0dd6f34955b60c. Examples are available [here](https://github.com/sagr4019/ResearchProject/blob/master/data-generation-and-validation/security-type-system/examples.txt).

**Comment by jan-christiansen** [comment]
* You don't have to store the operator because the AST abstracts from the syntax.
* An assignment is not an operator because an assignment does not yield a result.
* I fI am not mistaken `Str` is a variable. Please rename it.
* The left-hand side of an assignment should always be a variable. Therefore, the left-hand side should be represented by a simple string.

**Comment by jan-christiansen** [comment]
* You don't have to store the operator symbol, an AST is supposed to abstract from the concrete syntax.
* An assignment is not a operator. An operator yields a results like addition or comparisons.
* The left-hand side of an assignment is supposed to be a variable. Please just use a string for the left-hand side.
* Please add an example for sequential composition.
* `Str` is a variable if I am not mistaken, please rename it.

**Comment by MadeInGer** [comment]
>     * The left-hand side of an assignment should always be a variable. Therefore, the left-hand side should be represented by a simple string.

Yes, but according to https://github.com/sagr4019/ResearchProject/issues/110:

> Don't bother about type correctness

So I thought that we should implement the rules of grammar from the [paper](http://users.cis.fiu.edu/%7Esmithg/papers/jcs96.pdf) and put it into practice without changing the rules in the first step. So we're free to change the rules as long as we keep record about it in our paper?

**Comment by MadeInGer** [comment]
Some of my notes from the meeting
- rename 'Concat' to 'Seq'
- rename 'Do' to 'Body'
